### PR TITLE
[exporter/sumologicexporter]: adds support for tracing

### PR DESCRIPTION
**Pull request by sumo-drosiek**
**Description:**

adds support for tracing

It should be the last PR which moves code from Sumo Logic repository. I will compare the codebase after merging it

**Link to tracking Issue:**

#32315

**Testing:**

* unit tests

**Documentation:**

* Readme

### CODE REVIEW

**Comment by sumo-drosiek** [comment]
I hope this is the last one. I am going to be very happy about closing this saga 🤞

File: exporter/sumologicexporter/config.go [hunk 1]
```diff
@@ -37,6 +37,7 @@ type Config struct {
 	// Format to post logs into Sumo. (default json)
 	//   * text - Logs will appear in Sumo Logic in text format.
 	//   * json - Logs will appear in Sumo Logic in json format.
+	//   * otlp - Logs will be send in otlp format and will appear in Sumo Logic in text format.
```
Comment (swiatekm-sumo) [inline exporter/sumologicexporter/config.go:40]: This isn't correct. Rather, the way the logs are presented in Sumo will depend on the record attributes and the data type of Body. 
* If record attributes are empty, and the body is a string, we show plain text. 
* If record attributes are empty, and the body is a structured data type, we show the JSON equivalent of body.
* If they're not, we show a JSON object with the log body in the "body" key, and the rest of record attributes as the rest of the keys.

To be honest, this is complicated enough that it should really be a README section, or we can link to the Sumo documentation.

Comment (sumo-drosiek) [reply exporter/sumologicexporter/config.go]: I linked doucumentation in `README.md` and `config.go`

**Review by swiatekm — APPROVED** [review]

**Comment by sumo-drosiek** [comment]
cc @songy23 @dmitryax @andrzej-stencel

File: .chloggen/drosiek-tracing.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
+# Use this changelog template to create an entry for release notes.
+
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: breaking
```
Comment (songy23) [inline .chloggen/drosiek-tracing.yaml:4]: Why is it a breaking change?

File: exporter/sumologicexporter/exporter.go [hunk 3]
```diff
@@ -374,6 +397,26 @@ func (se *sumologicexporter) handleUnauthorizedErrors(ctx context.Context, errs
 	}
 }
 
+func (se *sumologicexporter) pushTracesData(ctx context.Context, td ptrace.Traces) error {
+	logsURL, metricsURL, tracesURL := se.getDataURLs()
+	sdr := newSender(
```
Comment (songy23) [inline exporter/sumologicexporter/exporter.go:402]: I'm not familiar with Sumo but why does it need to create a new sender for each trace batch, why not create a sender once and reuse it?

File: .chloggen/drosiek-tracing.yaml (hunk 2 above)
Comment (sumo-drosiek) [reply .chloggen/drosiek-tracing.yaml:4]:
```suggestion
change_type: enhancement
```

File: exporter/sumologicexporter/exporter.go (hunk 3 above)
Comment (sumo-drosiek) [reply exporter/sumologicexporter/exporter.go:402]: We used to have implementation which would required synchronisation before, but now I think we could reuse it 🤔

I created issue for it: https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/33105

**Review by songy23 — APPROVED** [review]

**Comment by sumo-drosiek** [comment]
cc: @collector-contrib-approvers

File: exporter/sumologicexporter/config.go [hunk 4]
```diff
@@ -46,6 +51,11 @@ type Config struct {
 	// Decompose OTLP Histograms into individual metrics, similar to how they're represented in Prometheus format
 	DecomposeOtlpHistograms bool `mapstructure:"decompose_otlp_histograms"`
 
+	// Traces related configuration
+	// The format of traces you will be sending, currently only otlp format is supported
+	TraceFormat TraceFormatType `mapstructure:"trace_format"`
```
Comment (crobert-1) [inline exporter/sumologicexporter/config.go:56]: Do you want to add a README entry for this option? 

Side note, if there's only one valid, is it worth having a config option? (I guess this depends on whether there are plans to support more formats, I'll defer to you and other code owners here as to usefulness.)

File: exporter/sumologicexporter/config.go [hunk 5]
```diff
@@ -118,35 +148,39 @@ type LogFormatType string
...
 	NoCompression configcompression.Type = ""
 	// MetricsPipeline represents metrics pipeline
 	MetricsPipeline PipelineType = "metrics"
 	// LogsPipeline represents metrics pipeline
 	LogsPipeline PipelineType = "logs"
+	// TracesPipeline represents traces pipeline
+	TracesPipeline PipelineType = "traces"
 	// defaultTimeout
-	defaultTimeout time.Duration = 5 * time.Second
+	defaultTimeout time.Duration = 30 * time.Second
```
[the remark this answers is not in the document]
Comment (sumo-drosiek) [reply exporter/sumologicexporter/config.go:173]: Updated

File: exporter/sumologicexporter/config.go (hunk 4 above)
Comment (sumo-drosiek) [reply exporter/sumologicexporter/config.go]: I just removed it. It seems we do not use it anywhere, so should be safe to remove. We can add it in the future if we need to

File: exporter/sumologicexporter/README.md [hunk 6]
```diff
@@ -99,7 +101,7 @@ exporters:
     decompose_otlp_histograms: {true, false}
 
     # timeout is the timeout for every attempt to send data to the backend,
-    # maximum connection timeout is 55s, default = 5s
+    # maximum connection timeout is 55s, default = 30s
```
Comment (crobert-1) [inline exporter/sumologicexporter/README.md:104]: Is this maximum connection timeout ever enforced or tested? It's unrelated to this PR, so no need to do anything now, just something to consider if it's not currently enforced.

**Review by crobert-1 — APPROVED** [review]

File: exporter/sumologicexporter/README.md (hunk 6 above)
Comment (sumo-drosiek) [reply exporter/sumologicexporter/README.md:104]: I don't think we test it
Created issue so we can keep track of it: https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/33151

**Pull request merged**